src/main/java/com/esri/samples/raster/raster_rendering_rule/RasterRenderingRuleSample.java

```diff
@@ -0,0 +1,198 @@
+      controlsVBox.getStyleClass().add("panel-region");
+
+      // create progress indicator
+      ProgressIndicator progressIndicator = new ProgressIndicator();
+      progressIndicator.setVisible(false);
+
+      // create drop down menu of Rendering Rules
+      renderingRuleInfoComboBox = new ComboBox<>();
+      renderingRuleInfoComboBox.setMaxWidth(260.0);
+      renderingRuleInfoComboBox.setConverter(new StringConverter<>() {
+        @Override
+        public String toString(RenderingRuleInfo renderingRuleInfo) {
+
+          return renderingRuleInfo != null ? renderingRuleInfo.getName() : "";
+        }
+
+        @Override
+        public RenderingRuleInfo fromString(String string) {
+          return null;
+        }
+      });
+
+      // create a label for the Rendering Rule info
+      Label renderingRuleInfoLabel = new Label("");
+      renderingRuleInfoLabel.setWrapText(true);
+
+      // add the ComboBox and Label to the controlsVBox
+      controlsVBox.getChildren().addAll(renderingRuleInfoComboBox, renderingRuleInfoLabel);
+
+      // create a Streets BaseMap
+      ArcGISMap map = new ArcGISMap(Basemap.createStreets());
+
+      // add the map to a new map view
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create an Image Service Raster as a raster layer and add to map
+      final ImageServiceRaster imageServiceRaster = new ImageServiceRaster(ImageServiceRasterUri);
+      final RasterLayer imageRasterLayer = new RasterLayer(imageServiceRaster);
+      map.getOperationalLayers().add(imageRasterLayer);
+
+      // add event listener to loading of Image Service Raster and wait until loaded
+      imageRasterLayer.addDoneLoadingListener(() -> {
+        if (imageRasterLayer.getLoadStatus() == LoadStatus.LOADED) {
+          // zoom to extent of the raster
+          mapView.setViewpointGeometryAsync(imageServiceRaster.getServiceInfo().getFullExtent());
+
+          // get the predefined rendering rules
+          List<RenderingRuleInfo> renderingRuleInfos = imageServiceRaster.getServiceInfo().getRenderingRuleInfos();
+
+          // populate the drop down menu with the rendering rule names
+          renderingRuleInfoComboBox.getItems().addAll(renderingRuleInfos);
+          
+          // listen to selection in the drop-down menu
+          renderingRuleInfoComboBox.getSelectionModel().selectedItemProperty().addListener(o -> {
+
+            // get the requested rendering rule info from the list
+            RenderingRuleInfo selectedRenderingRuleInfo = renderingRuleInfoComboBox.getSelectionModel().getSelectedItem();
+
+            // change the label text to the rendering rule info description
+            String renderingRuleInfoDescription = selectedRenderingRuleInfo.getDescription();
+            renderingRuleInfoLabel.setText("Rule Description: \n" + renderingRuleInfoDescription);
+
+            // clear previous rendering rule from all operational layers
```

Comment:
this line of code is clearing the operational layers from the map: i.e. it is removing the raster layer that you have placed within the operational layer each time. I'd use:

`// clear previous raster layer from the map's operational layers`